src/SharpLearning.Backend.Cntk.Test/CntkRawMnistTest.cs

```diff
@@ -20,7 +20,124 @@ public class CntkRawMnistTest
         const string DownloadPath = "MnistTest";
 
         [TestMethod]
-        public void MnistTrainTest()
+        public void MnistSimpleExampleTest()
+        {
+            // Set global data and device type. 
+            var dataType = CntkDataType.Float;
+            DeviceDescriptor device = DeviceDescriptor.UseDefaultDevice();
+            Trace.WriteLine("Using device: " + device.Type + " with data type: " + dataType);
+
+            // Define data.
+            var inputDimensions = 784;
+            var numberOfClasses = 10;
+
+            // Input variables denoting the features and label data.
+            Variable x = Variable.InputVariable(new int[] { inputDimensions }, dataType); ;
+            Variable y = Variable.InputVariable(new int[] { numberOfClasses }, dataType);
+
+            // Model Parameters.
+            Parameter w = new Parameter(new int[] { numberOfClasses, inputDimensions }, dataType, 0.0f, device, "W");
+            Parameter b = new Parameter(new int[] { numberOfClasses }, dataType, 0.0f, device, "B");
+            Function m = CNTKLib.Times(w, x); // variable and function are implicitly convertable
+            
+            // Linear Model.
+            Function model = m + b;
+
+            // Define loss and error metric.
+            Function loss = CNTKLib.CrossEntropyWithSoftmax(model, y);
+            Function errorMetric = CNTKLib.ClassificationError(model, y);
+
+            // Training config.
+            var minibatchSize = 64;
+            var minibatchIterations = 2000;
+
+            // Instantiate progress writers.
+            var trainingProgressOutputFreq = 100;
+
+            // Instantiate the trainer object to drive the model training.
+            var lr = new TrainingParameterScheduleDouble(0.01, 1);
+            Trainer trainer = Trainer.CreateTrainer(model, loss, errorMetric, 
+                new List<Learner>() { Learner.SGDLearner(model.Parameters(), lr) });
+
+            // Load train data.
+            var mnist = Mnist.Load(DownloadPath);
+            var numberOftrainingSamples = mnist.TrainImages.Length;
+            var readerTrain = mnist.GetTrainReader();
+
+            // Train model.
+            var inputMap = new Dictionary<Variable, Value>();
+            for (int i = 0; i < minibatchIterations; i++)
```

Comment:
@nietras Compared to the corresponding python example, the C# training loop is a lot more complicated due to the use of raw arrays and manual copying of batch data. It is possible to change the C# example to use the text minibatch source similar to the python example to make them more similar. However, this would make the cntk examples deviate more from the tensorflow examples.

Alternatively, we can consider changing the cntk python examples to load the raw images like the cntk c# examples.